website/docs/core-concepts/writing-code/workflows.md

```diff
@@ -2,155 +2,161 @@
 
-For example, the properties like `OnTextChanged` and `OnSubmit` of an input widget are considered async fields. You can use these properties to define an action or perform an [operation](../../reference/appsmith-framework/#functions).
+### Fields
 
-### **Use an Async Function Response in a Sync Field**
+Widgets provide fields that you can use to bind and display data or trigger an action. Based on this, the properties are categorized as either sync or async.
 
- <VideoEmbed host="youtube" videoId="yn_8gs5w04g" title="Using data from Async function in Synchronous Field" caption="Using data from Async function in Synchronous Field"/>  
+#### Sync fields
 
+Sync fields are properties that expect input or data in the properties pane. These fields are updated immediately when the data changes.
 
-Let's look at some examples to understand how you can use async functions in sync fields.
+![Input widget - Sync Fields](</img/Writing_Code__Creating_Workflows__Sync_Fields__Input_Widget.png>)
 
-#### **Use case**
+For example, when you add an Input widget to the canvas, properties such as `Max Characters`, `Regex`, `Error Message`, and others expect input and are considered as Sync Fields.
 
-You are fetching all the users and displaying information like First Name, Last Name, Email, and more on a page. You have a table widget to display the data.
+#### Async fields
 
-**Using an API**
+Async fields are properties that can trigger an action or perform an operation. They're updated as soon as the data is received.
 
-You have an API that fetches Users. You want to trigger the API execution, so you bind the API call to the widget, and the response generated will be shown in the table.
+![Input Widget - Async Fields](</img/Writing_Code__Creating_Workflows__Async_Fields__Input_Widget.png>)
 
-To add a Table widget; Navigate to `Explorer` >> Click on `Widgets` >> Search `table` >> Drag a table widget onto the canvas.
+For example, the properties like `OnTextChanged` and `OnSubmit` of an input widget are considered async fields. You can use these properties to define an [action](/reference/appsmith-framework/widget-actions) or perform and operation like [API or Query](/core-concepts/data-access-and-binding/querying-a-database#running-a-query) execution.
 
- <VideoEmbed host="youtube" videoId="iYZV9DPnugY" title="Using an API" caption="Using an API"/>  
+#### Async function data in a sync field
+When working with data in Appsmith, it's important to understand how to use async functions in sync fields. Asynchronous functions can be a powerful tool, but they can also be a source of errors if not used correctly. For more information, see [How to use data from async functions in sync fields](/learning-and-resources/how-to-guides/how-to-use-data-from-async-functions-in-sync-fields).
```

Comment:
@jnikhila i think you missed this . And I still feel we should add the video here so that the user can quickly go over and then if they want to read in detail they can go to the how to guide. You can keep the video there too